kdc/kerberos5.c

```diff
@@ -1816,6 +1840,21 @@ _kdc_as_rep(kdc_request_t r,
 	    if (pa) {
 		ret = pat[n].validate(r, pa);
 		if (ret != 0) {
+		    krb5_error_code  ret2;
+		    Key *ckey = NULL;
+		    /*
+		     * If there is a client key, send ETYPE_INFO{,2}
+		     */
+		    ret2 = _kdc_find_etype(context,
+					   config->preauth_use_strongest_session_key,
+					   TRUE, r->client, b->etype.val,
+					   b->etype.len, NULL, &ckey);
+		    if (ret2 == 0) {
```

Comment:
I think I shouldn't ignore error from _kdc_find_etype(), I'll change to:
```suggestion
		    if (ret2) {
		        ret = ret2;
		        goto out;
		    }
		    ret2 = get_pa_etype_info_both(0) ...
```